interfaces.go

```diff
@@ -67,8 +71,10 @@ type Permission interface {
 	SetPermissionProvider(PermissionProvider)
 }
 
+type PermissionProviderGetPermission func(permissionProvider PermissionProvider, role Role, permission string) Permission
```

Comment:
Name the type for the func that? I do think the interfaces need to be trimmed in some way... an my func types are obnoxious.